### PR TITLE
Breadcrumb component

### PR DESCRIPTION
**Pull request by spiropoulos94**
This pull request contains the Breadcrumb Component and it's own Styles

### CODE REVIEW

File: src/components/Breadcrumbs/index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "/visits": "All Visits",
+  "/patients": "Patients",
+  "/patients/details/test-id": "Patient Details",
+  "/patients/create-new": "Patient Create New",
+  "/patients/edit/test-id": "Patient Edit"
+}
+
+function BreadcrumbsComponent() {
+  return (
+    <Router>
```
Comment (nikoskleidis) [inline src/components/Breadcrumbs/index.js:23]: We are not supposed to have two Router components in our app. There should be only one and that one already exists in [App.js](https://github.com/SocialHackerCreteClass/FrodidApp-FrontEnd/blob/develop/src/App.js) file.
I think it is safe to delete the Router and render just the Breadcrumbs here

File: src/components/LoginStatus/index.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import React from "react"
 import PropTypes from "prop-types"
 import { useI18n } from "providers/I18n"
+import BreadcrumbsComponent from "../Breadcrumbs"
```
[the remark this answers is not in the document]
Comment (spiropoulos94) [reply src/components/LoginStatus/index.js:4]: yes you are right i was trying to get it being shown everywhere thats why i left it there. i had not intention of leaving it there

File: src/components/Menu/index.js [hunk 3]
```diff
@@ -31,18 +31,6 @@ function Menu() {
         <li>
           <Link to="/visits/edit/test-id">Edit a visit</Link>
         </li>
-        <li>
```
[the remark this answers is not in the document]
Comment (spiropoulos94) [reply src/components/Menu/index.js:34]: there was a duplicate entry of a link

File: src/components/Menu/index.js [hunk 4]
```diff
@@ -23,37 +23,25 @@ function Menu() {
           <Link to="/visits">All Visits</Link>
         </li>
         <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
+          <Link to="/visits/details/:id">View a single Visit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:26]: When we use a link, we use an instance of the url. The dynamic url as it is defined in Routes (and in breadcrumbs) is:` "/visits/details/:id"`
Possible instances of this url are:
`"/visits/details/asdrewt234234rsfsdjkj234"`
`"/visits/details/asdf"`
`"/visits/details/test-id"`

So `:id` is just a placeholder when we define the routes. When we actually use them, we need to provide an actual value to that placeholder

Comment (spiropoulos94) [reply src/components/Menu/index.js:26]: so ypu are suggesting to have a url that will end up like "/details:id" ? i mean the id will be joined on the details?

Comment (nikoskleidis) [reply src/components/Menu/index.js:26]: I didn't suggest changing any urls. I just said that this change needs to be reverted. 
The Link should be like this : `<Link to="/visits/details/test-id">View a single Visit</Link>`
Do you understand the difference between the configuration url `"/visits/details/:id"` and the instance url `"/visits/details/test-id"` of that configuration?

Comment (spiropoulos94) [reply src/components/Menu/index.js:26]: i am not completely sure and the reason is that if you suggest keeping a placeholder then why use /test-id instead of /:id ?

Comment (nikoskleidis) [reply src/components/Menu/index.js]: The url of the details page of a specific visit is `"/visits/details/:id"`. The layout of the page is the same for every visit that we are going to display. What changes is the data. The data are unique for each visit. 
So when we are listing the some visits urls on a table we are going to type something like this:
`<Link to="/visits/details/visit-id-1">Press here to see the details of visit 1</Link>`
`<Link to="/visits/details/visit-id-2">Press here to see the details of visit 2</Link>`
The id of the first visit is: **visit-id-1**.  We are going to use this id in order to retrieve the data of this visit from the server.
If we click on the second link we are going to be navigated to **http://localhost:3000/visits/details/visit-id-2**. The id of that visit is **visit-id-2**. 
So we notice that there is a pattern there. The part **/visits/details/** is always the same but the last part is dynamic and it is based on the visit id. 
This is why when we tell to react router that when you see `"/visits/details/:id"` display the VisitDetails component. 
Inside the VisitDetails we are going to extract the id from the url (eg "visit-id-1", "visit-id-2") and we are going to retrieve the data of that visit from the server.
So to sum up. We use `"/visits/details/:id"` on the react router (and breadcrumb) configuration in order to direct all similar urls to the VisitDetails component.
We use `<Link to="/visits/details/visit-id-2">Press here to see the details of visit 2</Link>` to navigate to the details of a specific visit. 

In this example I created a custom link `<Link to="/visits/details/test-id">View a single Visit</Link>` in order to demonstrate how the urls are going to look. If this was an actual id (that would exist on the server) it would display the details of the visit with _id = test-id_.
Saying something like `<Link to="/visits/details/:id">View a single Visit</Link>` doesn't fit here because that is a configuration url that should be specified at the react-router configuration

Comment (spiropoulos94) [reply src/components/Menu/index.js]: So if i get this right, the **`/visits/details/:id`** doesnt work because there is no :id page. the id is supposed to FIT to an already declared page and change it's content (because of the different data) Right?

Comment (nikoskleidis) [reply src/components/Menu/index.js]: Exactly :id is just a placeholder that is used during the configuration of the routing

File: package.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "bootstrap": "^4.5.0",
```
Comment (nikoskleidis) [inline package.json:9]: We will not be using bootstrap. Please remove this package

File: src/App.js [hunk 6]
```diff
@@ -16,9 +17,14 @@ function App() {
     localStorage.setItem("user is logged", loggedStatus)
   }, [loggedStatus])
 
+  //I have placed the breadcrumb component inside the App.js in order to be visible from everywhere and display changes.
```
Comment (nikoskleidis) [inline src/App.js:20]: We want it inside the logged in pages only. Breadcrumb is not displayed in the login or the register pages. 
You can place it in `src/Pages/index.js`

File: src/components/Menu/index.js [hunk 7]
```diff
@@ -23,37 +23,25 @@ function Menu() {
           <Link to="/visits">All Visits</Link>
         </li>
         <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
+          <Link to="/visits/details:id">View a single Visit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:26]: This change needs to be reverted as discussed

File: src/components/Menu/index.js [hunk 8]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
-        </li>
-        <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
-        </li>
-        <li>
-          <Link to="/visits/create-new">New visit</Link>
-        </li>
-        <li>
-          <Link to="/visits/edit/test-id">Edit a visit</Link>
+          <Link to="/visits/edit:id">Edit a visit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:32]: This change needs to be reverted as discussed

File: src/components/Menu/index.js [hunk 9]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
         </li>
         <li>
           <Link to="/patients">Patients</Link>
         </li>
         <li>
-          <Link to="/patients/details/test-id">Patient Details</Link>
+          <Link to="/patients/details:id">Patient Details</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:38]: This change needs to be reverted as discussed

File: src/components/Menu/index.js [hunk 10]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
         </li>
         <li>
           <Link to="/patients/create-new">Patient Create New</Link>
         </li>
         <li>
-          <Link to="/patients/edit/test-id">Patient Edit</Link>
+          <Link to="/patients/edit:id">Patient Edit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:44]: This change needs to be reverted as discussed

File: src/components/Menu/index.js [hunk 11]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
-        </li>
-        <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
-        </li>
-        <li>
-          <Link to="/visits/create-new">New visit</Link>
-        </li>
-        <li>
-          <Link to="/visits/edit/test-id">Edit a visit</Link>
+          <Link to="/visits/edit-id">Edit a visit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:32]: The config path is "/visits/edit/:id".
So the correct link here is "/visits/edit/edit-id"

File: src/components/Menu/index.js [hunk 12]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
         </li>
         <li>
           <Link to="/patients">Patients</Link>
         </li>
         <li>
-          <Link to="/patients/details/test-id">Patient Details</Link>
+          <Link to="/patients/details-id">Patient Details</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:38]: The config path is "/patients/details/:id".
So the correct link here is "/patients/details/details-id"

File: src/components/Menu/index.js [hunk 13]
```diff
@@ -23,37 +23,25 @@ function Menu() {
...
         </li>
         <li>
           <Link to="/patients/create-new">Patient Create New</Link>
         </li>
         <li>
-          <Link to="/patients/edit/test-id">Patient Edit</Link>
+          <Link to="/patients/edit-id">Patient Edit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:44]: The config path is "/patients/edit/:id".
So the correct link here is "/patients/edit/edit-id"

File: src/components/Menu/index.js [hunk 14]
```diff
@@ -23,37 +23,25 @@ function Menu() {
           <Link to="/visits">All Visits</Link>
         </li>
         <li>
-          <Link to="/visits/details/test-id">View a single Visit</Link>
+          <Link to="/visits/details-id">View a single Visit</Link>
```
Comment (nikoskleidis) [inline src/components/Menu/index.js:26]: The config path is "/visits/details/:id".
So the correct link here is "/visits/details/details-id"
